modules/combined/test/tests/optimization/compliance_sensitivity/3d_mbb.i

```diff
@@ -0,0 +1,238 @@
+    order = CONSTANT
+    initial_condition = ${vol_frac}
+  []
+  [Dc_elem]
+    family = MONOMIAL
+    order = CONSTANT
+    initial_condition = -1.0
+    [AuxKernel]
+      type = SelfAux
+      variable = Dc_elem
+      v = Dc
+      execute_on = 'TIMESTEP_END'
+    []
+  []
+  [mat_den_nodal]
+    family = L2_LAGRANGE
+    order = FIRST
+    initial_condition = ${vol_frac}
+    [AuxKernel]
+      type = SelfAux
+      execute_on = TIMESTEP_END
+      variable = mat_den_nodal
+      v = mat_den
+    []
+  []
+[]
+
+[Modules/TensorMechanics/Master]
+  [all]
+    strain = SMALL
+    add_variables = true
+    incremental = false
+  []
+[]
+[Kernels]
+  [diffusion]
+    type = FunctionDiffusion
+    variable = Dc
+    function = 0.15 # radius coeff
+  []
+  [potential]
+    type = Reaction
+    variable = Dc
+  []
+  [source]
+    type = CoupledForce
+    variable = Dc
+    v = sensitivity
+  []
+[]
+[BCs]
+  [no_x]
+    type = DirichletBC
+    variable = disp_y
+    boundary = hold_y
+    value = 0.0
+  []
+  [no_y]
+    type = DirichletBC
+    variable = disp_x
+    boundary = right
+    value = 0.0
+  []
+  [boundary_penalty]
+    type = ADRobinBC
+    variable = Dc
+    boundary = 'left top front back'
+    coefficient = 10
+  []
+  [boundary_penalty_right]
+    type = ADRobinBC
+    variable = Dc
+    boundary = 'right'
+    coefficient = 10
+  []
+[]
+[NodalKernels]
+  [push]
+    type = NodalGravity
+    variable = disp_y
+    boundary = push
+    gravity_value = -1
+    mass = 1
+  []
+[]
+
+[Materials]
+  [elasticity_tensor]
+    type = ComputeVariableIsotropicElasticityTensor
+    youngs_modulus = E_phys
+    poissons_ratio = poissons_ratio
+    args = 'mat_den'
+  []
+  [E_phys]
+    type = DerivativeParsedMaterial
+    # Emin + (density^penal) * (E0 - Emin)
+    expression = '${Emin} + (mat_den ^ ${power}) * (${E0}-${Emin})'
+    coupled_variables = 'mat_den'
+    property_name = E_phys
+  []
+  [poissons_ratio]
+    type = GenericConstantMaterial
+    prop_names = poissons_ratio
+    prop_values = 0.3
+  []
+  [stress]
+    type = ComputeLinearElasticStress
+  []
+  [dc]
+    type = ComplianceSensitivity
+    design_density = mat_den
+    youngs_modulus = E_phys
+    incremental = false
+  []
+[]
+
+[Preconditioning]
+  [smp]
+    type = SMP
+    full = true
+  []
+[]
+[UserObjects]
+  [update]
+    type = DensityUpdate
+    density_sensitivity = Dc_elem
+    design_density = mat_den
+    power = ${power}
+    volume_fraction = ${vol_frac}
+    execute_on = TIMESTEP_BEGIN
+    force_postaux = true
+  []
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = NEWTON
+  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
+  petsc_options_value = 'lu superlu_dist'
+  line_search = none
+  nl_abs_tol = 1e-4
+  l_max_its = 200
+  start_time = 0.0
+  dt = 1.0
+  num_steps = 2
+[]
+
+[Outputs]
+  [out]
+    type = CSV
+    execute_on = 'INITIAL TIMESTEP_END'
+  []
+  print_linear_residuals = false
+[]
+
+[Postprocessors]
```

Comment:
While not optimal, the postprocessors are good to avoid regression. I suggest you create a follow-on PR when you decide which specific postprocessors you'd like in your optimization inputs.